lib/active_admin/filters/forms.rb

```diff
@@ -57,7 +57,10 @@ def active_admin_filters_form_for(search, filters, options = {})
           filters.each do |attribute, opts|
             next if opts.key?(:if)     && !call_method_or_proc_on(self, opts[:if])
             next if opts.key?(:unless) &&  call_method_or_proc_on(self, opts[:unless])
-            opts[:input_html] = instance_exec(&opts[:input_html]) if opts[:input_html].is_a?(Proc)
+            if opts[:input_html].is_a?(Proc)
+              opts = opts.dup
```

Comment:
It would be better to avoid assigning the `opts` variable because it can be confused with the block variable with same name. Something like this would be a better approach:

```ruby
filters.each do |attribute, opts|
  next if opts.key?(:if)     && !call_method_or_proc_on(self, opts[:if])
  next if opts.key?(:unless) &&  call_method_or_proc_on(self, opts[:unless])

  filter_opts = opts.except(:if, :unless)
  filter_opts[:input_html] = instance_exec(&fitler_opts[:input_html]) if filter_opts[:input_html].is_a?(Proc)

  f.filter attribute, filter_opts
end
```

Reply:
Indeed. What you propose is much better. I have updated the code